commons/zenoh-protocol/src/core/endpoint.rs

```diff
@@ -310,6 +345,13 @@ impl fmt::Display for MetadataMut<'_> {
     }
 }
 
+#[cfg(feature = "defmt")]
```

Comment:
Can we use `derive(defmt::Format)`?

Reply:
I think `derive` would derive an implementation based on the fields of a struct, instead of "copying" the behaviour of `Display` (which is what we want here). One solution to avoid duplicating code would be using [this](https://docs.rs/defmt/latest/defmt/struct.Display2Format.html), but there will still be the problem of quite a lot boilerplate code. I'll continue investigating if there is a better solution.

Reply:
The solution I mentioned previously may no be suitable either. From `defmt`'s docs:

<img src="https://user-images.githubusercontent.com/22601842/211324274-0d212174-b461-4f74-9b98-4eab0ab34edb.png" alt="immagine" style="max-width: 100%;" width="600px">